components/admin_console/generated_setting.tsx

```diff
@@ -3,30 +3,27 @@
 
 import crypto from 'crypto';
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 
 import SetByEnv from './set_by_env';
 
-export default class GeneratedSetting extends React.Component {
-    static get propTypes() {
-        return {
-            id: PropTypes.string.isRequired,
-            label: PropTypes.node.isRequired,
-            placeholder: PropTypes.string,
-            value: PropTypes.string.isRequired,
-            onChange: PropTypes.func.isRequired,
-            disabled: PropTypes.bool.isRequired,
-            setByEnv: PropTypes.bool.isRequired,
-            disabledText: PropTypes.node,
-            helpText: PropTypes.node.isRequired,
-            regenerateText: PropTypes.node,
-            regenerateHelpText: PropTypes.node,
-        };
-    }
+type Props = {
+    id: string;
+    label: React.ReactNode;
+    placeholder?: string;
+    value: string;
+    onChange: (id: string, s: string) => void;
+    disabled: boolean;
+    setByEnv: boolean;
+    disabledText?: React.ReactNode;
+    helpText: React.ReactNode;
+    regenerateText: React.ReactNode;
```

Comment:
This should be optional

Reply:
Nevermind, this is going to be generated by the default props anyway.

Reply:
Yes. As far I have seen, the most common way to go about default props is to set them as needed (no `?`) and then on default. It may be slightly harder to read (at first glance you may think they are required fields that you must assign, as it happened here), but it is the current standard. Maybe it would be interesting to mark "defaulted" props with a comment, in order to improve legibility. (i.e. `regenerateText: React.ReactNode; //Default value`)